### ISSUE TITLE
Is it possible to allow `single_component_path_imports` for an individual `use`?

### ISSUE THREAD

**Issue by marshallpierce** [rust-lang/rust-clippy#8768]
### Summary

I use `rstest_reuse`, which [requires](https://docs.rs/rstest_reuse/0.3.0/rstest_reuse/#cavelets) being `use`d at the top level of the crate because of how its macros work. The `single_component_path_imports` lint (correctly) flags this, but I need it to be that way, and I can't figure out how to allow that lint for just that one `use`. I can disable the offending lint globally with `#!allow`, but I'd rather not do that as it's generally a valid lint.

### Reproducer

I tried this code:

```rust
#[cfg(test)]
use rstest_reuse;
```

I saw this happen:

```
error: this import is redundant
   --> src/lib.rs:100:1
    |
100 | use rstest_reuse;
    | ^^^^^^^^^^^^^^^^^ help: remove it entirely
    |
note: the lint level is defined here
   --> src/lib.rs:86:5
    |
86  |     warnings
    |     ^^^^^^^^
    = note: `#[deny(clippy::single_component_path_imports)]` implied by `#[deny(warnings)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
```

So I added an `allow`:

```rust
#[cfg(test)]
#[allow(clippy::single_component_path_imports)]
use rstest_reuse;
```

and now two lints are mad:

```
error: this import is redundant
   --> src/lib.rs:100:1
    |
100 | use rstest_reuse;
    | ^^^^^^^^^^^^^^^^^ help: remove it entirely
    |
note: the lint level is defined here
   --> src/lib.rs:86:5
    |
86  |     warnings
    |     ^^^^^^^^
    = note: `#[deny(clippy::single_component_path_imports)]` implied by `#[deny(warnings)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

error: useless lint attribute
  --> src/lib.rs:99:1
   |
99 | #[allow(clippy::single_component_path_imports)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(clippy::single_component_path_imports)]`
   |
   = note: `#[deny(clippy::useless_attribute)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute
```

### Version

```text
rustc 1.60.0 (7737e0b5c 2022-04-04)
binary: rustc
commit-hash: 7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c
commit-date: 2022-04-04
host: x86_64-apple-darwin
release: 1.60.0
LLVM version: 14.0.0
```

**Comment by ijackson** [comment]
This happened to me too, while trying to work around a FP (#9718)
